hikari/impl/entity_factory.py

```diff
@@ -1748,6 +1748,16 @@ def _deserialize_command_option(self, payload: data_binding.JSONObject) -> comma
         if raw_channel_types := payload.get("channel_types"):
             channel_types = [channel_models.ChannelType(channel_type) for channel_type in raw_channel_types]
 
+        name_localizations: typing.Mapping[str, str] = {}
+        if raw_name_localizations := payload.get("name_localizations"):
+            name_localizations = {locales.Locale(k): raw_name_localizations[k] for k in raw_name_localizations}
+
+        description_localizations: typing.Mapping[str, str] = {}
+        if raw_description_localizations := payload.get("description_localizations"):
+            description_localizations = {
+                locales.Locale(k): raw_description_localizations[k] for k in raw_description_localizations
```

Comment:
The default could be moved to the else clause in both of these

Reply:
Is this really needed? I feel like setting the default like this is better and more consistent with rest of the code.

Reply:
Would like to move away from that old way of doing it, as it is a bit wasteful

@FasterSpeeding thoughts?

Reply:
> Would like to move away from that old way of doing it, as it is a bit wasteful
> 
> @FasterSpeeding thoughts?

There's no new or "old" way of doing stuff, the approach has always been to set the default before the if case when the default is a singleton, otherwise set it in an else statement.

Reply:
So, is there a need to change it?

Reply:
> So, is there a need to change it?

From what snab said, if its a singleton (which a dict isnt), then set it as the default. In any other case, set it in the else.

So in this case, should be in the else